File: semana-1/url-to-short.rb

puts "Digite a url que deseja encurtar:"
# Ref https://www.rubyguides.com/2019/10/ruby-chomp-gets/
url = gets.chomp

hashUrl = ""
#Ref https://www.rubyguides.com/2015/05/working-with-files-ruby/
# Ref Open mode: https://www.rubydoc.info/stdlib/core/IO:initialize
File.open("urls.csv", "a+") do |f|
    # Ref https://ruby-doc.org/stdlib-3.1.1/libdoc/digest/rdoc/index.html
    
    require 'digest/sha1'

    # Ref https://ruby-doc.org/core-3.1.1/String.html#method-i-5B-5D
    hashUrl = Digest::SHA1.hexdigest(url)[0..7]
    f.write("#{hashUrl},#{url}\n")
end

puts "A sua url curta é: shorter.app/#{hashUrl}"

File: semana-1/url-to-short.js
const crypto = require("crypto")
const fs = require('fs');

const prefix = process.env.APP_WEBSITE || 'shorter.app'


//ref https://www.geeksforgeeks.org/node-js-hash-digest-method/
const createHash = (url) => crypto.createHash("sha256").update(url).digest("hex").substring(0,7)
//Ref https://nodejs.org/api/fs.html#fswritefilesyncfile-data-options
// Ref https://nodejs.org/api/fs.html#file-system-flags
const persistData = (url) =>  fs.writeFileSync("urls.csv", url, {'flag':'a'});



// Ref https://nodejs.dev/learn/accept-input-from-the-command-line-in-nodejs
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  
readline.question('Digite a url que deseja encurtar:', url => {
  readline.close();

  let digest = createHash(url)
  persistData(`${digest},${url}`)
  console.log(`A sua url curta é: ${prefix}/${digest}`)  
});